lib/inc/pxp-agent/modules/task.hpp

```diff
@@ -8,6 +8,7 @@
 #include <pxp-agent/util/purgeable.hpp>
 #include <pxp-agent/util/bolt_module.hpp>
 
+#include <cpp-pcp-client/util/thread.hpp>
```

Comment:
I don't _think_ this line should be required, since all the thread work is done within the modulecachedir class which already includes this header.